client/rest-high-level/src/test/java/org/elasticsearch/client/RequestConvertersTests.java

```diff
@@ -1834,7 +1936,14 @@ private static void setRandomIndicesOptions(Consumer<IndicesOptions> setter, Sup
             Map<String, String> expectedParams) {
 
         if (randomBoolean()) {
-            setter.accept(IndicesOptions.fromOptions(randomBoolean(), randomBoolean(), randomBoolean(), randomBoolean()));
+            // This caused failures in testMultiSearchTemplate's request equality assertion: 
+            // setter.accept(IndicesOptions.fromOptions(randomBoolean(), randomBoolean(), randomBoolean(), randomBoolean()));
+            
+            // See https://github.com/elastic/elasticsearch/issues/20105#issuecomment-265692775 
+            // The above notes the discrepancy between REST-configurable params and those in the Java client. 
+            // It proposes the solution below:
+            setter.accept(IndicesOptions.fromOptions(randomBoolean(), randomBoolean(), randomBoolean(), randomBoolean(), 
+                    SearchRequest.DEFAULT_INDICES_OPTIONS));
```

Comment:
good point, but I don't think that using the default options for search works for all the other API? At least not in theory. maybe we should allow to provide the default ones for each API?

Reply:
I found this "options" part of the code pretty confusing to be honest. Could do with a quick chat about what's going on here.

Reply:
`IndicesOptions` holds some options that can be set at REST (allow_no_indices, Ignore_unavailable, expand_wildcards) and some others that describe the behaviour of an API, which are hardcoded (on purpose). Unfortunately the transport client allows to set also the latter, with catastrophic results though :)

The point is that, for instance, the search API cannot work against closed indices, hence forbid closed indices is set to true in the default search api indices options, which means that closed indices are considered "unavailable". The default indices options matter as they hold default values, but also some values that cannot be overridden (unless you do it from the transport client at your own risk).

This is not a problem in the high-level REST client, as you can potentially set values like forbidClosedIndices etc. but they will be ignored, only the ones supported at REST will be converted to query_string params. That is what we check in these tests. I suspect that this change is needed because of your request comparison which should not be here but rather in the specific test around multiline serialization.